server/sonar-server/src/main/java/org/sonar/server/computation/issue/NewDebtAggregator.java

```diff
@@ -0,0 +1,121 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.issue;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.core.issue.DefaultIssue;
+import org.sonar.core.issue.db.IssueChangeDto;
+import org.sonar.core.issue.tracking.Tracking;
+import org.sonar.server.computation.component.Component;
+import org.sonar.server.computation.measure.Measure;
+import org.sonar.server.computation.measure.MeasureRepository;
+import org.sonar.server.computation.measure.MeasureVariations;
+import org.sonar.server.computation.metric.Metric;
+import org.sonar.server.computation.metric.MetricRepository;
+import org.sonar.server.computation.period.Period;
+import org.sonar.server.computation.period.PeriodsHolder;
+import org.sonar.server.db.DbClient;
+
+public class NewDebtAggregator extends IssueVisitor {
+
+  private final NewDebtCalculator calculator;
+  private final PeriodsHolder periodsHolder;
+  private final DbClient dbClient;
+  private final MetricRepository metricRepository;
+  private final MeasureRepository measureRepository;
+
+  private ListMultimap<String, IssueChangeDto> changesByIssueUuid = ArrayListMultimap.create();
+  private Map<Integer, DebtSum> sumsByComponentRef = new HashMap<>();
+  private DebtSum currentSum = null;
+
+  public NewDebtAggregator(NewDebtCalculator calculator, PeriodsHolder periodsHolder, DbClient dbClient,
+    MetricRepository metricRepository, MeasureRepository measureRepository) {
+    this.calculator = calculator;
+    this.periodsHolder = periodsHolder;
+    this.dbClient = dbClient;
+    this.metricRepository = metricRepository;
+    this.measureRepository = measureRepository;
+  }
+
+  @Override
+  public void beforeComponent(Component component, Tracking tracking) {
+    currentSum = new DebtSum();
+    sumsByComponentRef.put(component.getRef(), currentSum);
+    List<IssueChangeDto> changes = dbClient.issueChangeDao().selectChangelogOfNonClosedIssuesByComponent(component.getUuid());
+    for (IssueChangeDto change : changes) {
+      changesByIssueUuid.put(change.getIssueKey(), change);
+    }
+    for (Component child : component.getChildren()) {
+      DebtSum childSum = sumsByComponentRef.remove(child.getRef());
+      if (childSum != null) {
+        currentSum.add(childSum);
+      }
+    }
+  }
+
+  @Override
+  public void onIssue(Component component, DefaultIssue issue) {
+    // TODO filter unresolved
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)